File: command/command/document/DocumentItemFactory.cs
using command.storage;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace command.document
{
	public class DocumentItemFactory : IDocumentItemFactory
	{
		private readonly IFileStorage _fileStorage;

		public DocumentItemFactory(IFileStorage fileStorage) => _fileStorage = fileStorage ?? throw new ArgumentNullException(nameof(fileStorage));

		public IImage CreateImage(string path, uint width, uint height)
		{
			return new Image(_fileStorage.Add(path), width, height);
		}

		public IParagraph CreateParagraph(string text)
		{
			return new Paragraph(text);
		}
	}
}

File: command_java/Document/src/document/documentItem/ItemsCollection/DocumentItemFactory.java
package document.documentItem.ItemsCollection;

import document.documentItem.image.IImage;
import document.documentItem.image.Image;
import document.documentItem.image.Size;
import document.documentItem.paragraph.IParagraph;
import document.documentItem.paragraph.Paragraph;
import storage.IFileStorage;

import java.io.IOException;

public class DocumentItemFactory implements IDocumentItemFactory, AutoCloseable {

    public DocumentItemFactory(IFileStorage storage){
        _storage = storage;
    }

    @Override
    public void close() throws Exception {
        _storage.close();
    }

    @Override
    public IImage CreateImage(String path, Size size) throws IOException {
        return new Image(_storage.Add(path), size);
    }

    @Override
    public IParagraph CreateParagraph(String text) {
        return new Paragraph(text);
    }

    private final IFileStorage _storage;
}
